_field-types/flat-object.md

```diff
@@ -0,0 +1,218 @@
+---
+layout: default
+title: Flat object
+nav_order: 43
+has_children: false
+parent: Object field types
+grand_parent: Supported field types
+---
+
+# Flat object field type
+
+In OpenSearch, you don't have to specify a mapping before indexing documents. If you don't specify a mapping, OpenSearch uses [dynamic mapping]({{site.url}}{{site.baseurl}}/field-types/mappings#dynamic-mapping) to map every field and its subfields in the document automatically. When you ingest documents such as logs, you may not know every field's subfield name and type in advance. In this case, dynamically mapping all new subfields can quickly lead to a "mapping explosion", where the growing number of fields may degrade the performance of your cluster. 
+
+Flat object solves this problem by treating the entire JSON object as a string. Subfields within the JSON object are accessible using standard dot path notation but they are not indexed for fast lookup.
+
+The maximum field value length in the dot notation is 2<sup>24</sup> &minus; 1.
+{: .note}
+
+Flat objects provide the following advantages:
```

Comment:
In the preceding paragraphs, "flat object" is used in singular form. Is there a distinction between the usage above and the usage in lines 19-33? For example, in line 14, are we referring to the field type? If so, I suggest writing "The flat object field type..."

Reply:
Changed. Thanks!